appinventor/components/src/com/google/appinventor/components/runtime/util/GingerbreadUtil.java

```diff
@@ -127,7 +127,8 @@ public static void resolveNFCIntent(Intent intent, NearField nfc){
         byte[] payload = msgs[0].getRecords()[0].getPayload();
         //the substring chops off the two language encoding bits at the beginning
         String message = new String(payload).substring(3);
-        nfc.TagRead(message);
+        Tag tag = intent.getParcelableExtra(NfcAdapter.EXTRA_TAG);
```

Comment:
It probably doesn't matter too much at this point, but the EXTRA_TAG was added in SDK 10, whereas the rest of the NFC code was added in SDK 9. So for the random person who might be running a phone with Android 2.3 `tag` will be null. If we want to be pedantic about null safety, something like:

```java
String tagId = "";
if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.GINGERBREAD_MR1) {
  Tag tag = intent.getParcelableExtra(NfcAdapter.EXTRA_TAG);
  tagId = NearField.toHexString(tag.getId());
}
nfc.TagRead(tagId, message);
```

would be more appropriate. Then once we update the minSDK to 14 we can inline things.

Reply:
Uh, didn't notice it. I guess we missed it as we were already working with way higher API levels.